pkg/cli/image/extract/extract.go

```diff
@@ -358,7 +361,15 @@ func (o *ExtractOptions) Run() error {
 	}
 
 	stopCh := make(chan struct{})
-	defer close(stopCh)
+	linuxConversionWarning := ""
+	defer func() {
+		close(stopCh)
+		setLinuxConversionWarningOnce.Do(func() {
```

Comment:
Yeah, I was just trying to prevent the message from appearing twice. If you are okay, with it appearing twice, I can remove this. As a user, I would not want the duplicated error messages.

I had the boolean flag prior to this, but it seems to add more complexity than the current solution. It would require a mutex since the flag could be accessed by both channels, right? This implementation seems simpler. I didn't add any additional defers; I just added the `Do.Once` function to the current defer.

Reply:
I agree, it makes sense to prevent duplicate messages from user point of view. I'd use boolean without managing it with mutexes.